module/zfs/zfs_ioctl.c

```diff
@@ -3258,22 +3311,30 @@ zfs_ioc_clone(const char *fsname, nvlist_t *innvl, nvlist_t *outnvl)
 {
 	int error = 0;
 	nvlist_t *nvprops = NULL;
+	nvlist_t *hidden_args = NULL;
+	dsl_crypto_params_t *dcp = NULL;
 	char *origin_name;
 
 	if (nvlist_lookup_string(innvl, "origin", &origin_name) != 0)
 		return (SET_ERROR(EINVAL));
 	(void) nvlist_lookup_nvlist(innvl, "props", &nvprops);
+	(void) nvlist_lookup_nvlist(innvl, ZPOOL_HIDDEN_ARGS, &hidden_args);
```

Comment:
We could do that. It was originally combined to get the semantics of Oracle's `zfs key -K` in an atomic way. I personally kind of like the atomicity, rather than having strange "half-successful" clones which userspace needs to deal with as best it can.